File: JavaScript/7js 4/solve_ssr.js
/*
|--------------------------------------------------------------------------
| Simple string reversal
|--------------------------------------------------------------------------
|
*/

function solve(str) {
   let x = str.split(" ").join("").split("").reverse();
   str.split("").map((v, i) => {
      if (v == " ") x.splice(i, 0, " ");
   });
   return x.join("");
}


function solve(str) {
   let arr = [...str].filter((x) => x != " ");
   return str.replace(/\S/g, (_) => arr.pop());
}


function solve(str) {
   let s = "";
   for (let i = str.length - 1; i >= 0; i--) {
      if (str[i] != " ") s += str[i];
      if (str[s.length] == " ") s += str[s.length];
   }
   return s;
}


console.log(solve("codewars"), "srawedoc");
console.log(solve("your code"), "edoc ruoy");
console.log(solve("your code rocks"), "skco redo cruoy");
console.log(solve("i love codewars"), "s rawe docevoli");

File: Python/7py 4/solve_ssr.py
"""
|--------------------------------------------------------------------------
| Simple string reversal
|--------------------------------------------------------------------------
|
"""

def solve(s):
   x = reversed(s.replace(" ", ""))
   return "".join(i if i == " " else next(x) for i in s)


def solve(s):
   rev = list(s.replace(" ", "")[::-1])
   for index, item in enumerate(s):
      if item == " ":
         rev.insert(index, item)
   return "".join(rev)


def solve(s):
   r = list(s.replace(" ", ""))
   return "".join(i if i == " " else r.pop() for i in s)


print(solve("codewars"),"srawedoc")
print(solve("your code"),"edoc ruoy")
print(solve("your code rocks"),"skco redo cruoy")
print(solve("i love codewars"),"s rawe docevoli")